browser/net/brave_site_hacks_network_delegate_helper.cc

```diff
@@ -50,12 +55,91 @@ bool ApplyPotentialReferrerBlock(std::shared_ptr<BraveRequestInfo> ctx) {
   return false;
 }
 
+bool ApplyPotentialQueryStringFilter(std::shared_ptr<BraveRequestInfo> ctx) {
+  DCHECK_CURRENTLY_ON(BrowserThread::IO);
+  if (!ctx->request_url.has_query()) {
+    return false;
+  }
+
+  std::string new_query = ctx->request_url.query();
+
+  bool modified = false;
+  for (auto tracker : query_string_trackers) {
```

Comment:
Maybe we can simplify it by using something from `url_util.h`? I.e. iterate with `QueryIterator` and copy only needed key/values

Reply:
Sadly, no. At least not if we want to minimize the changes to the query string and avoid rewriting it as much as possible: https://github.com/brave/brave-core/pull/3239#issuecomment-524073918

Reply:
Sorry, I've missed that comment. I din't exactly understand what do you mean by minimizing changes, could you give some examples? If the existing Chromium solution still keeps things correct, my vote is for using it even if it is not 100% optimal

Reply:
It means we're rewriting the query strings and losing many parts of the original one. For example, we would rewrite `?foo&&bar=yes&gclid=1234` as `?foo=&bar=yes`. We end up with a lot more changes than just stripping out the `gclid` value.

If I use the built-in parser, then none of the test cases under https://github.com/brave/brave-core/pull/3239/files#diff-80e007808537c31b7bc01c9acde4469cR324 can be made to work.

Reply:
Thanks, I see now. Though not sure if some extra modifications are much important if we stay aligned with the spec.
Anyway, if we decide to go for the custom parser, it's worth to explain this situation and extract the parsing code to at least a separate function or, even better, separate file.